From: Juan Balderas <juan.balderas.0@gmail.com>
Date: Tue, 26 Feb 2019 13:52:57 -0500
Subject: Handle URLs when given as command line parameters.

Bug-Debian: 612470
Last-Update: 2011-02-10

Applications use %U in .desktop file should be able to open multiple
files either in the form of paths or URLs.
---
 po/POTFILES.skip |  1 +
 src/file.c       | 27 ++++++++++++++++++++++++++-
 2 files changed, 27 insertions(+), 1 deletion(-)
 create mode 100644 po/POTFILES.skip

diff --git a/po/POTFILES.skip b/po/POTFILES.skip
new file mode 100644
index 0000000..294a3e5
--- /dev/null
+++ b/po/POTFILES.skip
@@ -0,0 +1 @@
+.pc/debian-612470-handle-urls.patch/src/file.c
diff --git a/src/file.c b/src/file.c
index 51e2012..f7f4839 100644
--- a/src/file.c
+++ b/src/file.c
@@ -133,11 +133,30 @@ file_save_dialog ( void )
 
 
 gboolean	
-file_open ( const gchar * filename )
+file_open ( const gchar * fname )
 {
 	gboolean	ok		=	FALSE;
 	GError 		*error	= NULL;
 	GdkPixbuf 	*pixbuf	= NULL;
+	gchar 		*filename = NULL;
+	GFile		*file	= NULL;
+
+	filename = g_strdup (fname);
+
+	file = g_file_new_for_uri (filename);
+	if( G_IS_FILE (file) )
+	{
+		gchar *p = NULL;
+
+		p = g_file_get_path ( file);
+		if(p)
+		{
+			g_free (filename);
+			filename = p;
+		}
+		g_object_unref(file);
+	}
+
 	pixbuf = gdk_pixbuf_new_from_file (filename, &error);
 	if (pixbuf != NULL)
 	{
@@ -178,6 +197,12 @@ file_open ( const gchar * filename )
 		g_free (basename);
 		g_free (message);
 	}
+
+	if(filename)
+	{
+		g_free (filename);
+	}
+
 	g_object_unref (pixbuf);
 	return ok;
 }
